integrationtests/self/http_test.go

```diff
@@ -1068,4 +1069,118 @@ var _ = Describe("HTTP tests", func() {
 			"Unannounced": {"Surprise!"},
 		})))
 	})
+
+	It("graceful shutdown successfully with no timeout", func() {
```

Comment:
I'm not sure I understand what this test actually tests. Let's make this more intuitive.

1. Remove `/fast`.
2. Remove the timer from `/slow`. Instead, block on a channel.
3. Call `CloseGracefully`. Make sure that it blocks.
4. Close the channel.
5. Assert that `CloseGracefully` returns.

Reply:
You can't close a channel when `CloseGracefully` is blocked, it has to be done before that. Http handler has no way of knowing whether a graceful shutdown started.